src/ngraph_backend_manager.h

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+// The backend manager class is a singelton class that interfaces with the
+// bridge to provide necessary backend
+
+#ifndef NGRAPH_TF_BRIDGE_BACKEND_MANAGER_H_
+#define NGRAPH_TF_BRIDGE_BACKEND_MANAGER_H_
+
+#include <mutex>
+#include <ostream>
+#include <vector>
+
+#include "ngraph/ngraph.hpp"
+#include "ngraph/runtime/backend_manager.hpp"
+#include "ngraph_log.h"
+#include "tensorflow/core/lib/core/errors.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+struct Backend {
+  unique_ptr<ng::runtime::Backend> backend_ptr;
+  mutex backend_mutex;
+};
+
+class BackendManager {
+ public:
+  // Returns the backend name currently set
+  static string GetCurrentlySetBackendName() {
+    NGRAPH_VLOG(5) << "Getting backend";
```

Comment:
Can we say: `Reading backend name`

Reply:
This was just a debug log. Removing this :)